src/main/java/com/emc/pravega/perf/PravegaPerfTest.java

```diff
@@ -259,131 +314,278 @@ private static void parseCmdLine(String[] args) {
+                                 ReaderGroupConfig.builder()
+                                                  .stream(Stream.of(scope, stream))
+                                                  .build());
+              readerGroup = readerGroupManager.getReaderGroup(stream);
+              return readerGroup;
+      }     
+
+}
 
-        final EventStreamWriter<String> producer;
-        private final int producerId;
-        private final int eventsPerSec;
-        private final int secondsToRun;
-        private final boolean isTransaction;
 
-        WriterWorker(int sensorId, int eventsPerSec, int secondsToRun, boolean isTransaction, boolean isRandomKey,
-                     ClientFactory factory) {
+class WriterWorker implements Callable<Void> {
+      final EventStreamWriter<String> producer;
+      private final int producerId;
+      private final int eventsPerSec;
+      private final int secondsToRun;
+      private final int messageSize; 
+      private final long StartTime;
+      private final boolean isRandomKey;
+      private final PerfStats stats;
+
+      WriterWorker(int sensorId, int eventsPerSec, int secondsToRun,
+                   boolean isRandomKey, int messageSize,  long start,
+                   ClientFactory factory, PerfStats stats,  String streamName) {
             this.producerId = sensorId;
             this.eventsPerSec = eventsPerSec;
             this.secondsToRun = secondsToRun;
-            this.isTransaction = isTransaction;
+            this.StartTime = start;
+            this.stats = stats;
+            this.isRandomKey = isRandomKey; 
+            this.messageSize = messageSize;      
             this.producer = factory.createEventWriter(streamName,
                     new JavaSerializer<String>(),
                     EventWriterConfig.builder().build());
 
-        }
+      }
 
-        /**
-         * This function will be executed in a loop and time behavior is measured.
-         * @return A function which takes String key and data and returns a future object.
-         */
-        BiFunction<String, String, CompletableFuture> sendFunction() {
+      /**
+       * This function will be executed in a loop and time behavior is measured.
+       * @return A function which takes String key and data and returns a future object.
+       */
+      BiFunction<String, String, CompletableFuture> sendFunction() {
             return  ( key, data) -> producer.writeEvent(key, data);
-        }
+      }
 
-        /**
-         * Executes the given method over the producer with configured settings.
-         * @param fn The function to execute.
-         */
-        void runLoop(BiFunction<String, String, CompletableFuture> fn) {
+      /**
+        * Executes the given method over the producer with configured settings.
+        * @param fn The function to execute.
+        */
+      void runLoop(BiFunction<String, String, CompletableFuture> fn) throws Exception {
```

Comment:
Please throw the specific type of exception that you need.

Reply:
yes, now the Exceptions are no more generic , except in the main.. all the other places , the excpetions are specific now